packages/api/src/store/asset-table.ts

```diff
@@ -53,4 +53,19 @@ export default class AssetTable extends Table<WithID<Asset>> {
     }
     return assets[0];
   }
+
+  async getBySourceId(id: string): Promise<WithID<Asset>> {
+    const query = [
+      sql`asset.data->'source'->>'id' = ${id}`,
+      sql`asset.data->>'deleted' IS NULL`,
+    ];
+    const [assets] = await this.find(query, {
+      limit: 2,
+      order: "coalesce((asset.data->'createdAt')::bigint, 0) ASC",
+    });
+    if (!assets || assets.length < 1) {
+      return null;
+    }
+    return assets[0];
```

Comment:
Just calling out one unintended side-effect that may happen here - if two users create an asset with the same source CID, the first asset will have all views registered under it. I don't think this is a blocking issue, but something we'll need to iterate on.

Reply:
Yeah good call out!